backpack/core/derivatives/basederivatives.py

```diff
@@ -289,73 +393,315 @@ def weight_jac_t_mat_prod(self, module, g_inp, g_out, mat, sum_batch=True):
+    ) -> Tensor:
         raise NotImplementedError
 
 
-class BaseLossDerivatives(BaseDerivatives):
+class BaseLossDerivatives(BaseDerivatives, ABC):
     """Second- order partial derivatives of loss functions."""
 
     # TODO Add shape check
-    def sqrt_hessian(self, module, g_inp, g_out):
-        """Symmetric factorization ('sqrt') of the loss Hessian."""
-        self.check_2nd_order_make_sense(module, g_inp, g_out)
+    def sqrt_hessian(
+        self, module: Module, g_inp: Tuple[Tensor], g_out: Tuple[Tensor]
+    ) -> Tensor:
+        """Symmetric factorization ('sqrt') of the loss Hessian.
+
+        Args:
+            module: module to perform derivatives on
+            g_inp: input gradients
+            g_out: output gradients
+
+        Returns:
+            square root of hessian
+        """
+        self._check_2nd_order_make_sense(module, g_out)
         return self._sqrt_hessian(module, g_inp, g_out)
 
-    def _sqrt_hessian(self, module, g_inp, g_out):
+    def _sqrt_hessian(
+        self, module: Module, g_inp: Tuple[Tensor], g_out: Tuple[Tensor]
+    ) -> Tensor:
         raise NotImplementedError
 
     # TODO Add shape check
-    def sqrt_hessian_sampled(self, module, g_inp, g_out, mc_samples=1):
-        """Monte-Carlo sampled symmetric factorization of the loss Hessian."""
-        self.check_2nd_order_make_sense(module, g_inp, g_out)
+    def sqrt_hessian_sampled(
+        self,
+        module: Module,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mc_samples: int = 1,
+    ) -> Tensor:
+        """Monte-Carlo sampled symmetric factorization of the loss Hessian.
+
+        Args:
+            module: module to perform derivatives on
+            g_inp: input gradients
+            g_out: output gradients
+            mc_samples: number of monte carlo samples. Defaults to 1.
+
+        Returns:
+            square root of hessian
+        """
+        self._check_2nd_order_make_sense(module, g_out)
         return self._sqrt_hessian_sampled(module, g_inp, g_out, mc_samples=mc_samples)
 
-    def _sqrt_hessian_sampled(self, module, g_inp, g_out, mc_samples=1):
+    def _sqrt_hessian_sampled(
+        self,
+        module: Module,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mc_samples: int = 1,
+    ) -> Tensor:
         raise NotImplementedError
 
     @shape_check.make_hessian_mat_prod_accept_vectors
     @shape_check.make_hessian_mat_prod_check_shapes
-    def make_hessian_mat_prod(self, module, g_inp, g_out):
+    def make_hessian_mat_prod(
+        self, module: Module, g_inp: Tuple[Tensor], g_out: Tuple[Tensor]
+    ) -> Tensor:
         """Multiplication of the input Hessian with a matrix.
 
         Return a function that maps mat to H * mat.
+
+        Args:
+            module: module to perform derivatives on
+            g_inp: input gradients
+            g_out: output gradients
+
+        Returns:
+            hessian matrix product
```

Comment:
Replace with line 626

Reply:
Then the type hints for return value were wrong as well.